src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -25,11 +25,303 @@
+                break;
+            case Attributes::EnhancedColorMode::Id:
+                attIdx = ColorControlEFS::kEnhancedColorMode;
+                break;
+            default:
+                return CHIP_ERROR_INVALID_ARGUMENT;
+                break;
+            }
+
+            attributeBuffer[attIdx] = decodePair.attributeValue;
+        }
+
+        ReturnErrorOnFailure(pair_iterator.GetStatus());
+
+        // Switch to the mode saved in the scene
+        if (SupportsColorMode(endpoint, static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kEnhancedColorMode])))
+        {
+            ColorControlServer::Instance().handleModeSwitch(
+                endpoint, static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kEnhancedColorMode]));
+        }
+        else
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+
+        uint16_t transitionTime10th = static_cast<uint16_t>(timeMs / 100);
+
+#ifdef EMBER_AF_PLUGIN_COLOR_CONTROL_SERVER_HSV
+        ColorControlServer::ColorHueTransitionState * colorHueTransitionState =
+            ColorControlServer::Instance().getColorHueTransitionState(endpoint);
+        ColorControlServer::Color16uTransitionState * colorSaturationTransitionState =
+            ColorControlServer::Instance().getSaturationTransitionState(endpoint);
+        colorSaturationTransitionState->finalValue = static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kCurrentSaturation]);
+        colorHueTransitionState->finalEnhancedHue  = static_cast<uint16_t>(attributeBuffer[ColorControlEFS::kEnhancedCurrentHue]);
+#endif
+#ifdef EMBER_AF_PLUGIN_COLOR_CONTROL_SERVER_XY
+        ColorControlServer::Color16uTransitionState * colorXTransitionState =
+            ColorControlServer::Instance().getXTransitionState(endpoint);
+        ColorControlServer::Color16uTransitionState * colorYTransitionState =
+            ColorControlServer::Instance().getYTransitionState(endpoint);
+        colorXTransitionState->finalValue = static_cast<uint16_t>(attributeBuffer[ColorControlEFS::kCurrentX]);
+        colorYTransitionState->finalValue = static_cast<uint16_t>(attributeBuffer[ColorControlEFS::kCurrentY]);
+#endif
+#ifdef EMBER_AF_PLUGIN_COLOR_CONTROL_SERVER_TEMP
+        ColorControlServer::Color16uTransitionState * colorTempTransitionState =
+            ColorControlServer::Instance().getTempTransitionState(endpoint);
+        colorTempTransitionState->finalValue = static_cast<uint16_t>(attributeBuffer[ColorControlEFS::kColorTemperatureMireds]);
+#endif
+
+        // Set Loop Scene Attributes and start loop if scene stored active loop
+        Attributes::ColorLoopDirection::Set(endpoint, static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kColorLoopDirection]));
+        Attributes::ColorLoopTime::Set(endpoint, static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kColorLoopTime]));
+        if (attributeBuffer[ColorControlEFS::kColorLoopActive] == 1)
```

Comment:
Shouldn't this also be conditioned on this endpoint's server supporting the color loop feature?

Reply:
Added check on feature.